### ISSUE TITLE
Documentation unclear on how to have EXIF rotated rotated

### ISSUE THREAD

**Issue by joallard** [blueimp/JavaScript-Load-Image#57]
I'm using `loadImage(<file>, <function>, {orientation: true})`, however I can't get the image to get rotated back into the right position.

**Comment by jurgengeelen** [comment]
Can this function work by passing an url?

`function loadImageOrientation('https://location/image.jpg', callback, options)
    loadImage.parseMetaData(file, function (data) {
        if (data.exif) {
            options.orientation = data.exif.get('Orientation');
        }
        loadImage(file, callback, options);
    });
}`

By fetching the request it gives 'Promises: pending'.

**Comment by blueimp** [comment]
The `loadImageOrientation ` function example above only accepts a `File` or `Blob` object.

Loading different types of objects (URL/File/Blob) can be more complex based on the browser support you're aiming at and is covered by the main library file:
https://github.com/blueimp/JavaScript-Load-Image/blob/master/js/load-image.js

If you only support modern browsers, you could simply use [URL.createObjectURL](https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL).

**Comment by jurgengeelen** [comment]
Why does the code below not work to automatically rotate images?

>`    <div id="img">
    </div>
    <script src="js/load-image.all.min.js"></script>
    <script type="text/javascript">
	var url = 'https://test.com/test.jpeg'
	loadImage(
	url,
	function (img) {
		document.body.appendChild(img);
	},
	{
		maxWidth: 600,
		canvas:true,
		orientation:true
	});
    </script>`

**Comment by blueimp** [comment]
Can you provide a demo and provide a sample image with EXIF data that fails to be rotated?

Here's a working minimal example ([demo on JSFiddle](https://jsfiddle.net/n2b9qtLo/)):
```html
<script src="https://blueimp.github.io/JavaScript-Load-Image/js/load-image.all.min.js"></script>
<script>
  loadImage(
    'https://kartik-v.github.io/bootstrap-fileinput-samples/samples/Exif-Portrait-3.jpg',
    function (img) { document.body.appendChild(img) },
    { orientation: true }
  )
</script>
```

**Comment by blueimp** [comment]
Btw. my original answer to @joallard is not correct anymore, since I added the auto-orientation option in https://github.com/blueimp/JavaScript-Load-Image/commit/7e3304bcb40d8d4647340e3f2f7457807a279c06.

The orientation option is also documented in the [options config](https://github.com/blueimp/JavaScript-Load-Image/blob/master/README.md#options), so I'll close this issue for now.

Feel free to post additional comments if something should be unclear in the documentation.

**Comment by jurgengeelen** [comment]
Hello Sebastian,

It won't work yet. The photo does not want to rotate automatically.
In the attachment is the file 'index_test.html' which I used.
I have executed the file with the code here:
https://github.com/blueimp/JavaScript-Load-Image/tree/7e3304bcb40d8d4647340e3f2f7457807a279c06
I used the photo of you.
Do you know what I am doing wrong?

With best regards,
Jurgen

Op di 17 dec. 2019 om 13:23 schreef Sebastian Tschan <
notifications@github.com>:

**Comment by blueimp** [comment]
Unfortunately, GitHub doesn't include attachments when replying via Email, so I could not access your example file.
Could you enter its contents in the HTML field on [JSFiddle](https://jsfiddle.net/) instead, clicking on Save and sharing the resulting URL?

**Comment by jurgengeelen** [comment]
Here it is:

<!DOCTYPE html>
<html lang="en">
  <head>
<meta http-equiv="X-UA-Compatible" content="IE=edge" />
<meta charset="utf-8" />
  </head>
  <body>
    <div id="img">
    </div>
    <script src="js/load-image.all.min.js"></script>
    <script type="text/javascript">
  loadImage(
    'file://C:/Temp/JavaScript-Load-Image-Test/Exif-Portrait-3.jpg',
    function (img) {
document.body.appendChild(img)
},
    {
orientation: true
}
  )
    </script>
  </body>
</html>

Op vr 20 dec. 2019 om 06:15 schreef Sebastian Tschan <
notifications@github.com>:

**Comment by blueimp** [comment]
Alright, the problem is not your code, but the URL of the image file you are loading:
> 'file://C:/Temp/JavaScript-Load-Image-Test/Exif-Portrait-3.jpg'

You will likely see an error message similar to this one in the browser console:
> Fetch API cannot load file://C:/Temp/JavaScript-Load-Image-Test/Exif-Portrait-3.jpg. URL scheme must be "http" or "https" for CORS request.

If you load the image via `http(s)` instead (e.g. `'https://kartik-v.github.io/bootstrap-fileinput-samples/samples/Exif-Portrait-3.jpg'`), it should work.

The reason for this behavior is that image meta data (like EXIF orientation) can only be accessed if the image can be loaded as File object, which requires support for the [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) for images loaded via URL.
And since the Fetch API doesn't support `file:` URLs, this only works via `http(s)`.

**Comment by jurgengeelen** [comment]
I have pasted the url into it but the photo does not want to rotate
automatically. I tried it in Edge and Firefox.

<!DOCTYPE html>
<html lang="en">
  <head>
<meta http-equiv="X-UA-Compatible" content="IE=edge" />
<meta charset="utf-8" />
  </head>
  <body>
    <div id="img">
    </div>
    <script src="js/load-image.all.min.js"></script>
    <script type="text/javascript">
  loadImage(
    '
https://kartik-v.github.io/bootstrap-fileinput-samples/samples/Exif-Portrait-3.jpg
',
    function (img) {
document.body.appendChild(img)
},
    {
orientation: true
}
  )
    </script>
  </body>
</html>

Op vr 20 dec. 2019 om 11:26 schreef Sebastian Tschan <
notifications@github.com>:

**Comment by blueimp** [comment]
Could you please report if you see the correctly oriented image [on this page](https://fiddle.jshell.net/n2b9qtLo/show/) in your selected browsers?

That page is the same that is shown in the [JSFiddle demo](https://jsfiddle.net/n2b9qtLo/) I linked to earlier and uses the same code to automatically adjust the image orientation.

In my tests with latest Safari, Firefox and Chrome on macOS Catalina and latest Edge on Windows 10 the image is always oriented correctly (it will not work in Internet Explorer).

**Comment by jurgengeelen** [comment]
This example works in Edge.
However, I have a photo that does not work.
It should also work for me in Explorer because the Windows Webcontrol does
not support Edge

Op vr 20 dec. 2019 om 14:59 schreef Sebastian Tschan <
notifications@github.com>:

**Comment by jurgengeelen** [comment]
This is the link to the photo
https://photos.app.goo.gl/BkqHw1ZUnWWTPJ3LA

Op za 21 dec. 2019 om 09:13 schreef Jurgen Geelen <jurgengeelen@gmail.com>:

**Comment by blueimp** [comment]
The file you provided unfortunately does not contain any EXIF data and has already been correctly oriented - this is likely due to the upload to Google Photos.

I've just added support to parse meta data for images loaded via URL for Internet Explorer: https://github.com/blueimp/JavaScript-Load-Image/commit/f1a1080b82247bb1bf6a255125a4832101368b8d
It's available in the new release version `2.25.0`.

**Comment by jurgengeelen** [comment]
It works in Explorer. Awesome!
Only with large photos (3042 pixels, 5mb) does it not work in my index.html
yet.
In the index.html of yours, where you choose file upload, it works.
Here is the link of the photo:
https://photos.app.goo.gl/sUAA1q9QeYX24WqH8

Op za 21 dec. 2019 om 13:58 schreef Sebastian Tschan <
notifications@github.com>:
